cub/cub/detail/rfa.cuh

```diff
@@ -0,0 +1,950 @@
+  {
+    return *reinterpret_cast<uint64_t*>(&x);
+  }
+  /// Get the bit representation of a const float
+  _CCCL_DEVICE _CCCL_FORCEINLINE static uint32_t get_bits(const float& x)
+  {
+    return ::cuda::std::bit_cast<uint32_t>(x);
+  }
+  /// Get the bit representation of a const double
+  _CCCL_DEVICE _CCCL_FORCEINLINE static uint64_t get_bits(const double& x)
+  {
+    return ::cuda::std::bit_cast<uint64_t>(x);
+  }
+
+  /// Return primary vector value const ref
+  _CCCL_DEVICE _CCCL_FORCEINLINE const ftype& primary(int i) const
+  {
+    if (FOLD <= _CUB_RFA_MAX_JUMP_)
+    {
+      switch (i)
+      {
+        case 0:
+          if (FOLD >= 1)
+          {
+            return data[0];
+          }
+          [[fallthrough]];
+        case 1:
+          if (FOLD >= 2)
+          {
+            return data[1];
+          }
+          [[fallthrough]];
+        case 2:
+          if (FOLD >= 3)
+          {
+            return data[2];
+          }
+          [[fallthrough]];
+        case 3:
+          if (FOLD >= 4)
+          {
+            return data[3];
+          }
+          [[fallthrough]];
+        case 4:
+          if (FOLD >= 5)
+          {
+            return data[4];
+          }
+          [[fallthrough]];
+        default:
+          return data[FOLD - 1];
+      }
+    }
+    else
+    {
+      return data[i];
+    }
+  }
+
+  /// Return carry vector value const ref
+  _CCCL_DEVICE _CCCL_FORCEINLINE const ftype& carry(int i) const
+  {
+    if (FOLD <= _CUB_RFA_MAX_JUMP_)
+    {
+      switch (i)
+      {
+        case 0:
+          if (FOLD >= 1)
+          {
+            return data[FOLD + 0];
+          }
+          [[fallthrough]];
+        case 1:
+          if (FOLD >= 2)
+          {
+            return data[FOLD + 1];
+          }
+          [[fallthrough]];
+        case 2:
+          if (FOLD >= 3)
+          {
+            return data[FOLD + 2];
+          }
+          [[fallthrough]];
+        case 3:
+          if (FOLD >= 4)
+          {
+            return data[FOLD + 3];
+          }
+          [[fallthrough]];
+        case 4:
+          if (FOLD >= 5)
+          {
+            return data[FOLD + 4];
+          }
+          [[fallthrough]];
+        default:
+          return data[2 * FOLD - 1];
+      }
+    }
+    else
+    {
+      return data[FOLD + i];
+    }
+  }
+
+  /// Return primary vector value ref
+  _CCCL_DEVICE _CCCL_FORCEINLINE ftype& primary(int i)
+  {
+    const auto& c = *this;
+    return const_cast<ftype&>(c.primary(i));
+  }
+
+  /// Return carry vector value ref
+  _CCCL_DEVICE _CCCL_FORCEINLINE ftype& carry(int i)
+  {
+    const auto& c = *this;
+    return const_cast<ftype&>(c.carry(i));
+  }
+
+#ifdef DISABLE_ZERO
+  _CCCL_DEVICE _CCCL_FORCEINLINE static constexpr bool is_zero_v(const ftype)
+  {
+    return false;
+  }
+#else
+  _CCCL_DEVICE _CCCL_FORCEINLINE static constexpr bool is_zero_v(const ftype x)
+  {
+    return x == 0.0;
+  }
+#endif
```

Comment:
We have `[[maybe_unused]]` available now, so please move the `#ifdef DISABLED_ZERO` into the function call to reduce redundancy